posts/2023-09-26-spring-boot-3-instant-on.adoc

```diff
@@ -0,0 +1,215 @@
+
+For more information about Liberty InstantOn see the link:/blog/2023/06/29/rapid-startup-instanton.html[How to package your cloud-native Java application for rapid startup] blog and link:https://openliberty.io/docs/latest/instanton.html[Faster startup for containerized applications with Open Liberty InstantOn] in the Open Liberty docs.
+
+== Spring Boot support for checkpoint/restore
+
+The Spring Framework version 6.1 release will integrate with link:https://docs.spring.io/spring-framework/reference/6.1/integration/checkpoint-restore.html[JVM checkpoint/restore] by using the link:https://github.com/CRaC/org.crac[org.crac] project in order to allow capable systems to reduce the startup times of Spring-based Java applications. With the Liberty InstantOn 23.0.0.10-beta version a new `crac-1.3` feature can be configured to provide an implementation of the link:https://javadoc.io/doc/org.crac/crac/latest/index.html[org.crac API] that integrates with Liberty InstantOn. This allows Spring-based applications, including Spring Boot applications, to be deployed with Liberty InstantOn in order to achieve rapid startup times.
+
+== Production-ready Liberty container images
+
+New link:https://www.redhat.com/en/blog/introducing-red-hat-universal-base-image[Universal Base Image] container images are uploaded to the link:https://github.com/OpenLiberty/ci.docker/blob/main/docs/icr-images.md[IBM Container Registry] for each new release of Liberty. Starting with the 23.0.0.6 Liberty release, the Liberty UBI container images include the necessary prerequisites to checkpoint your applications with Liberty InstantOn. And now, starting with the 23.0.0.10-beta release, the UBI beta container image also includes the prerequisites to checkpoint your Spring Boot 3.2-based applications.
+
+This includes an implementation of the link:https://javadoc.io/doc/org.crac/crac/latest/index.html[`org.crac`] APIs with the Liberty beta feature `crac-1.3`. The `crac-1.3` feature along with the Spring Framework version 6.1 support for `org.crac` will allow you to checkpoint your Spring-based applications with Liberty InstantOn in order to achieve rapid startup times.
+
+The Liberty container images make it easy to produce InstantOn applications that are ready to deploy into production. An important benefit of using Liberty InstantOn is the ability to do a checkpoint of the application process inside the container without requiring the root user to be running the application process in the container. It is important, from a security perspective, to not run the application process in the container as the root user. This allows you to link:https://openliberty.io/docs/latest/instanton.html#_deploying_an_instanton_application_to_kubernetes_services[deploy your InstantOn container images] to existing Kubernetes services like AWS EKS and Azure EKS.
```

Comment:
```suggestion
The Liberty container images make it easy to develop InstantOn applications that are ready to deploy into production. An important benefit of using Liberty InstantOn is the ability to do a checkpoint of the application process inside the container without requiring the root user to be running the application process in the container. It is important, from a security perspective, to avoid running the application process in the container as the root user. This allows you to link:https://openliberty.io/docs/latest/instanton.html#_deploying_an_instanton_application_to_kubernetes_services[deploy your InstantOn container images] to existing Kubernetes services like AWS EKS and Azure EKS.
```